### PR TITLE
Refactor and bugfixes: Preparing for next release...

### PR DESCRIPTION
**Pull request by rilango**
Refactor visualization component to improve code quality. Additionally following
bugs are fixed:
- Rendering gradiant colored(on selection of molecular prop) scatter plot
- Broken reclustering after implementing multi-gpu version
- Following MoI on recluter and reload event
- Incorrect molecules filtered on point selection
- Remove fixed color list

Unit testcases:
- Test the ability to use pandas and cudf dataframe as input to gpu wf class
- Varify the columns returned from dao

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,13 +17,8 @@ RUN /opt/conda/envs/rapids/bin/pip install \
...
-
-# misc python packages
-RUN conda install -n rapids pywget
-
-# plotly
-RUN conda install -n rapids -c plotly plotly=4.9.0
+    autopep8 \
+    chembl_webresource_client
 
 RUN /opt/conda/envs/rapids/bin/pip install --ignore-installed --upgrade \
```
Comment (mlgill) [inline Dockerfile:23]: Should we fully remove the code for the cddd model? That would include tensorflow and model download in the Dockerfile, plus the embedding class.

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+progressbar2==3.53.1
+tables==3.6.1
+sqlalchemy==1.3.20
+openpyxl==3.0.6
+tabulate==0.8.7
+autopep8==1.5.4
+protobuf==3.14.0
+chembl_webresource_client==0.10.2
+dask_ml==1.8.0
+git+https://github.com/jrwnter/cddd.git
```
Comment (mlgill) [inline requirements.txt:16]: This would be removed as well if we decide to remove the embeddings.

File: setup/cuchem_rapids_0.17.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - plotly
+dependencies:
+  - rapids=0.17
+  - python=3.7
+  - cudatoolkit=11.0
+  - rdkit==2020.09.1
+  - plotly==4.9.0
+  - pip
+  - pip:
+    - tensorflow-gpu==1.15.5
```
Comment (mlgill) [inline setup/cuchem_rapids_0.17.yml:16]: Remove if we remove the embedding model.

File: setup/cuchem_rapids_0.17.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    - tables==3.6.1
+    - sqlalchemy==1.3.20
+    - openpyxl==3.0.6
+    - tabulate==0.8.7
+    - autopep8==1.5.4
+    - protobuf==3.14.0
+    - chembl_webresource_client==0.10.2
+    - dask_ml==1.8.0
+    - matplotlib==3.3.4
+    - git+https://github.com/jrwnter/cddd.git
```
Comment (mlgill) [inline setup/cuchem_rapids_0.17.yml:32]: And this

**Review by mlgill — APPROVED** [review]
I've made a couple of comments regarding whether or not we should remove the learned embeddings code (the cddd git repo) which is also the basis of the tensorflow requirement. 

I have not yet tested this branch -- please do not merge until I've confirmed that everything runs on my end. Thanks.

File: Dockerfile (hunk 1 above)
Comment (rilango) [reply Dockerfile]: At the moment that is the only alternate embedding function/model (apart from Morgan Fingerprint) to talk about. Though not useful, it highlights the ability to plugin new embedding models. Lets remove it once we have a better replacement.

File: requirements.txt (hunk 2 above)
Comment (rilango) [reply requirements.txt]: same as above.

**Pull request merged**